aleph-client/src/pallets/staking.rs

```diff
@@ -0,0 +1,382 @@
+
+        self.send_tx(tx, status).await
+    }
+
+    async fn validate(
+        &self,
+        validator_commission_percentage: u8,
+        status: TxStatus,
+    ) -> anyhow::Result<BlockHash> {
+        let tx = api::tx().staking().validate(ValidatorPrefs {
+            commission: Perbill(
+                SPerbill::from_percent(validator_commission_percentage as u32).deconstruct(),
+            ),
+            blocked: false,
+        });
+
+        self.send_tx(tx, status).await
+    }
+
+    async fn payout_stakers(
+        &self,
+        stash_account: AccountId,
+        era: EraIndex,
+        status: TxStatus,
+    ) -> anyhow::Result<BlockHash> {
+        let tx = api::tx().staking().payout_stakers(stash_account, era);
+
+        self.send_tx(tx, status).await
+    }
+
+    async fn nominate(
+        &self,
+        nominee_account_id: AccountId,
+        status: TxStatus,
+    ) -> anyhow::Result<BlockHash> {
+        let tx = api::tx()
+            .staking()
+            .nominate(vec![MultiAddress::Id(nominee_account_id)]);
+
+        self.send_tx(tx, status).await
+    }
+
+    async fn chill(&self, status: TxStatus) -> anyhow::Result<BlockHash> {
+        let tx = api::tx().staking().chill();
+
+        self.send_tx(tx, status).await
+    }
+
+    async fn bond_extra_stake(
+        &self,
+        extra_stake: Balance,
+        status: TxStatus,
+    ) -> anyhow::Result<BlockHash> {
+        let tx = api::tx().staking().bond_extra(extra_stake);
+
+        self.send_tx(tx, status).await
+    }
+}
+
+#[async_trait::async_trait]
+impl StakingSudoApi for RootConnection {
+    async fn force_new_era(&self, status: TxStatus) -> anyhow::Result<BlockHash> {
+        let call = Staking(force_new_era);
+
+        self.sudo_unchecked(call, status).await
+    }
+
+    async fn set_staking_config(
+        &self,
+        min_nominator_bond: u128,
+        min_validator_bond: u128,
+        max_nominator_count: Option<u32>,
+        max_validator_count: Option<u32>,
+        status: TxStatus,
+    ) -> anyhow::Result<BlockHash> {
+        let call = Staking(set_staking_configs {
+            min_nominator_bond: Set(min_nominator_bond),
+            min_validator_bond: Set(min_validator_bond),
+            max_nominator_count: match max_nominator_count {
+                None => ConfigOp::Noop,
+                Some(m) => Set(m),
+            },
+            max_validator_count: match max_validator_count {
+                None => ConfigOp::Noop,
+                Some(m) => Set(m),
+            },
+            chill_threshold: ConfigOp::Noop,
+            min_commission: ConfigOp::Noop,
+        });
+        self.sudo_unchecked(call, status).await
+    }
+}
+
+#[async_trait::async_trait]
+impl StakingRawApi for Connection {
+    async fn get_stakers_storage_keys(
+        &self,
+        era: EraIndex,
+        at: Option<BlockHash>,
+    ) -> Vec<StorageKey> {
+        let key_addrs = api::storage().staking().eras_stakers_root();
```

Comment:
we want to get storage keys for all stakers for given era. It cant be done differently since this is double map (you can only access it via eras_stakers with both era and account_id)

Reply:
thanks for explanation